src/directLine.ts

```diff
@@ -398,10 +405,33 @@ export class DirectLine implements IBotConnection {
         return once ? obs.take(1) : obs;
     }
 
+    setConnectionStatusFallback(
+        connectionStatusFrom: ConnectionStatus,
+        connectionStatusTo: ConnectionStatus,
+        maxAttempts = 5
+     ) {
+        let attempts = 0;
+        let currStatus = null;
+        return (status: ConnectionStatus): ConnectionStatus => {
+            if (status === connectionStatusFrom && currStatus === status) {
```

Comment:
How about doing two-if on one line?

```suggestion
            if (status === connectionStatusFrom && currStatus === status && attempts >= maxAttempts - 1) {
```